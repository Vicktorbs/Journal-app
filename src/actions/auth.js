import { firebase, googleAuthProvider } from "../firebase/firebase_config";
import { types } from "../types/types"

export const startLoginWithEmailPassword = (email, password) => {
    return (dispatch) => {
        setTimeout(() => {
            dispatch(login(12345, 'Juan'))
        }, 3500);
    }
}

export const startGoogleLogin = () => {
    return (dispatch) => {
        firebase.auth().signInWithPopup(googleAuthProvider).then(({ user }) => {
            dispatch(login(user.uid, user.displayName))
        })
    }
}

export const login = (uid, displayName) => {
    return {
        type: types.login,
        payload: {
            uid,
            displayName
        }
    }
}