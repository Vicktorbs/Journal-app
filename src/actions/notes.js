import Swal from "sweetalert2";
import { db } from "../firebase/firebase_config";
import { loadNotes } from "../helpers/loadNotes";
import { types } from "../types/types";

export const startNewNote = () => {
    return async(dispatch, getState) => {
        const uid = getState().auth.uid;

        const newNote = {
            title: '',
            body: '',
            date: new Date().getTime()
        }

        const documentRef = await db.collection(`${ uid }/journal/notes`).add(newNote);
        // console.log(documentRef);
        dispatch(activeNote(documentRef.id, newNote));
        // funcionalidad personalizada, checar
        dispatch(startLoadingNotes(uid));
    }
}

export const activeNote = (id, note) => ({
    type: types.notesActive,
    payload: {
        id,
        ...note
    }
})

export const startLoadingNotes = (uid) => {
    return async(dispatch) => {
        const notes = await loadNotes(uid);
        dispatch(setNotes(notes))
    }
}

export const setNotes = (notes) => ({
    type: types.notesLoad,
    payload: notes
})

export const startSaveNote = (note) => {
    return async(dispatch, getState) => {
        const uid = getState().auth.uid;

        if (!note.url) {
            delete note.url
        }

        const noteToFirestore = { ...note };
        delete noteToFirestore.id;

        await db.doc(`${ uid }/journal/notes/${ note.id }`).update(noteToFirestore);

        dispatch(refresNote(note.id, noteToFirestore));
        Swal.fire('Save', note.title, 'success')
    }
}

export const refresNote = (id, note) => ({
    type: types.notesUpdated,
    payload: {
        id,
        note: {
            id,
            ...note
        }
    }
})