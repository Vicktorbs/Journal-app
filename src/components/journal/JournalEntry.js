import React from 'react'

export const JournalEntry = () => {
    return (
        <div className="journal__entry pointer">
            <div 
                className="journal__entry-picture"
                style={{
                    backgroundSize: 'cover',
                    backgroundImage: 'url(https://ninjinanimehome.files.wordpress.com/2019/07/3e8ca-lupin2biii2bpart2bv2b255bsub-espa25c325b1ol255d255bmega-mf-gd255d255bhd-fullhd255d255bonline255d2bsoanimesitehd2.jpg)'
                }}
            ></div>
            <div className="journal__entry-body">
                <p className="journal__entry-title">
                    A new day
                </p>
                <p className="journal__entry-content">
                    loremeufndikf
                </p>
            </div>
            <div className="journal__entry-date-box">
                <span>Monday</span>
                <h3>28</h3>
            </div>
        </div>
    )
}
