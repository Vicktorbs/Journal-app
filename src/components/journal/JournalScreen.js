import React from 'react'

export const JournalScreen = () => {
    return (
        <div className="journal__main-countent">
            <h1>JournalScreen works!</h1>
        </div>
    )
}
