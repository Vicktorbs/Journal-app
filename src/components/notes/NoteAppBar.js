import React from 'react'

export const NoteAppBar = () => {
    return (
        <div className="notes_appbar">
            <span>25 de Abril 2021</span>

            <div>
                <button className="btn">
                    Picture
                </button>
                <button className="btn">
                    Save
                </button>
            </div>
        </div>
    )
}
