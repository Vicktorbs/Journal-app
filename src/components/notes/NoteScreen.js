import React from 'react'
import { NoteAppBar } from './NoteAppBar'

export const NoteScreen = () => {
    return (
        <div className="notes__main-content">
            <NoteAppBar />

            <div className="notes__content">
                <input type="text" placeholder="Some awesome title" className="notes__title-input" />

                <textarea placeholder="Whant happened today?" className="notes_text-area"></textarea>

                <div className="notes__images">
                    <img alt="Upload" src="https://i.pinimg.com/originals/c3/72/2c/c3722cfbfbe16b617bc55c547bb72691.jpg" />
                </div>
            </div>

        </div>
    )
}
