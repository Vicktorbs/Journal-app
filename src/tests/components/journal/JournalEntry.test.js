import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'; 
import thunk from 'redux-thunk';
import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {}

let store = mockStore(initialState);
store.dispatch = jest.fn()

const note = { 
    id: 10,
    date: 0,
    title: 'Hola note',
    body: 'Mundo note', 
    url: 'htttps://somewhere.com/foto.jpg'
}

const wrapper = mount(
    <Provider store={ store }>
        <JournalEntry { ...note } />
    </Provider>
)

describe('Test on component <JournalEntry />', () => {
    
    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should active the note', () => {
        wrapper.find('.journal__entry').prop('onClick')();
        expect(store.dispatch).toHaveBeenCalled();
        expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, { ...note }));
    })
    
})
