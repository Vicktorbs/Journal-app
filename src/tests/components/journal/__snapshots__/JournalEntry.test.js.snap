// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on component <JournalEntry /> should show itself correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn"
  onClick={[Function]}
>
  <div
    className="journal__entry-picture"
    style={
      Object {
        "backgroundImage": "url(htttps://somewhere.com/foto.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Hola note
    </p>
    <p
      className="journal__entry-content"
    >
      Mundo note
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h3>
      31st
    </h3>
  </div>
</div>
`;
